python-package/lightgbm/basic.py

```diff
@@ -1853,6 +1853,115 @@ def free_network(self):
+                    else:
+                        feature_name = t['split_feature']
+                else:
+                    feature_name = None
+                return feature_name
+
+            # Create the node record, and populate universal data members
+            node = {'tree_index': tree_index,
+                    'node_depth': node_depth,
+                    'node_index': _get_node_index(t, tree_index),
+                    'left_child': None,
+                    'right_child': None,
+                    'parent_index': parent_node,
+                    'split_feature': _get_split_feature(t, feature_names),
+                    'split_gain': None,
+                    'threshold': None,
+                    'decision_type': None,
+                    'missing_direction': None,
+                    'missing_type': None,
+                    'leaf_weight': None,
+                    'internal_count': None}
+
+            # Update values to reflect node type (leaf or split)
+            if _is_split_node(t):
+                node['left_child'] = _get_node_index(t['left_child'], tree_index)
+                node['right_child'] = _get_node_index(t['right_child'], tree_index)
+                node['split_gain'] = t['split_gain']
+                node['threshold'] = t['threshold']
+                node['decision_type'] = t['decision_type']
+                node['missing_direction'] = 'left' if t['default_left'] else 'right'
+                node['missing_type'] = t['missing_type']
+                node['internal_count'] = t['internal_count']
+            else:
+                node['leaf_weight'] = t['leaf_value']
+                node['internal_count'] = t['leaf_count']
+
+            return node
+
+        def tree_dict_to_node_list(t, node_depth=1, tree_index=None,
+                                   feature_names=None, parent_node=None):
+
+            node = create_node_record(t,
+                                      node_depth=node_depth,
+                                      tree_index=tree_index,
+                                      feature_names=feature_names,
+                                      parent_node=parent_node)
+
+            res = [node]
+
+            if _is_split_node(t):
+                # traverse the next level of the tree
+                children = ['left_child', 'right_child']
+                for child in children:
+                    subtree_list = tree_dict_to_node_list(
+                        t[child],
+                        node_depth=node_depth + 1,
+                        tree_index=tree_index,
+                        feature_names=feature_names,
+                        parent_node=node['node_index'])
+                    # In tree format, "subtree_list"is a list of node records (dicts), and we
+                    # add node to the list.
+                    res.extend(subtree_list)
+            return res
+
+        m_dict = self.dump_model()
```

Comment:
`model_dict` for better readability. Or what did you mean under the `m` letter?

Reply:
Done.